File: numIdenticalPairs.java
import java.util.*;
// Approach 1- HashMap
class Solution { // TC O(n) SC O(n)
    public int numIdenticalPairs(int[] nums) {
        Map<Integer, Integer> map=new HashMap<>();
        int ans=0;
        for(int num: nums){
            ans+=map.getOrDefault(num, 0);
            map.put(num, map.getOrDefault(num, 0)+1);
        }
        return ans;
    }
}

class numIdenticalPairs{
    public static void main(String[] args) {
        int[] nums={1,2,3,1,1,3};
        Solution s=new Solution();
        System.out.println(s.numIdenticalPairs(nums));
    }
}

File: numIdenticalPairs.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

// Approach 1- HashMap
class Solution { // TC O(n) SC O(n)
public:
    int numIdenticalPairs(vector<int>& nums) {
        unordered_map<int, int> map;
        int ans=0;
        for(int num: nums){
            ans+=map[num];
            map[num]++;
        }
        return ans;
    }
};

int main(){
    Solution s;
    vector<int> nums={1,2,3,1,1,3};
    cout << s.numIdenticalPairs(nums) << endl;
    return 0;
}